src/main/java/teammates/logic/core/FeedbackSessionsLogic.java

```diff
@@ -1110,6 +1110,29 @@ private StringBuilder getRowsOfPossibleRecipientsInCsvFormat(
         }
         return sessionsToSendEmailsFor;
     }
+    
+    /**
+     * Criteria: must be published, publishEmail must be enabled and
+     * resultsVisibleTime must be custom.
+     * 
+     * @return returns a list of sessions that require automated emails to be
+     *         sent as they are published
+     */
+    public List<FeedbackSessionAttributes> getFeedbackSessionsWhichNeedAutomatedUnpublishedEmailsToBeSent() {
+        List<FeedbackSessionAttributes> sessions =
+                fsDb.getFeedbackSessionsWithUnsentUnpublishedEmail();
+        List<FeedbackSessionAttributes> sessionsToSendEmailsFor =
+                new ArrayList<FeedbackSessionAttributes>();
+
+        for (FeedbackSessionAttributes session : sessions) {
+            // automated emails are required only for custom unpublish times
+            if (!session.isPublished() && session.isPublishedEmailEnabled()
```

Comment:
Hmm, it seems a bit awkward and confusing if `sentUnpublishedEmail` can be true even when no 'unpublished email' has been sent. I suspect we will need to change our approach here. The logic on `sentUnpublishedEmail` is not very clear, and it seems very coupled with `sentPublishedEmail`, so it might be possible to just rely on that?

Can we simply add code to `InstructorFeedbackUnpublishAction`? We make it such that if emails have been sent to inform students that the session is published, then we will send the `unpublished emails` to the students.